### ISSUE TITLE
doesn't use more that 12 cpus

### ISSUE THREAD

**Issue by shailysingh1975** [fireice-uk/xmr-stak-cpu#462]
I have a two Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, which provides total 48 logical processors(screenshot attached). I can see that the "xmr-stak-cpu" uses only 12 processors as i get best hashrate when i configure 12 cpu_threads in the config file. any number greater than 12 degrades the throughput and I get lowest with all 48 threads in the configuration.  I'm attaching screenshot of the processor and the total processor count for your review and the hashrate with various cpu_thread configuration. Please advise. I have purchased this server in the cloud and unable to utilize the full capacity, which is waste of money. please help.

Thank You in advance
Shailender
[ETNPerfomanceIssue.docx](https://github.com/fireice-uk/xmr-stak-cpu/files/1603243/ETNPerfomanceIssue.docx)

**Comment by JayWalker512** [comment]
This is because the Cryptonight algorithm does not scale linearly with core count, it depends on the amount of cache you have. That CPU has 30M cache, and each thread needs at minimum 2M for good performance. So 30/2 = 15 threads maximum. You only have 12 cores, so running additional threads on your "virtual cores" degrades performance as you saw. 

Please upgrade to [xmr-stak](https://github.com/fireice-uk/xmr-stak) for better performance and more configuration options. You may be able to improve your hash rate by running a few 2x or 3x type threads to utilize the full 30M cache, but you will not get benefit from going over 12 threads with this CPU model.

**Comment by shailysingh1975** [comment]
Thank you. There are 2 CPUs with 12 cores each in the machine and each have dedicated cache.  So if I understood above explanation, I have total 60M cache I should be able to double my performance.

**Comment by JayWalker512** [comment]
That is correct so long as you can ensure that the threads are running on the appropriate CPUs, ie. 12 on one and 12 on the other as opposed to 24 on one with the other idle.
